lib/WasmReader/WasmModule.cpp

```diff
@@ -33,8 +33,64 @@ WasmModule::WasmModule(Js::ScriptContext* scriptContext, byte* binaryBuffer, uin
     m_reader->InitializeReader();
 }
 
+uint32
+WasmModule::GetMaxFunctionIndex() const
+{
+    return UInt32Math::Add(GetImportCount(), GetWasmFunctionCount());
+}
+
+Wasm::WasmSignature*
+WasmModule::GetFunctionSignature(uint32 funcIndex) const
+{
+    FunctionIndexTypes::Type funcType = GetFunctionIndexType(funcIndex);
+    if (funcType == FunctionIndexTypes::Invalid)
```

Comment:
I did it this way, because I wanted to call `NormalizeFunctionIndex` which can throw if the type is invalid and the error message would be more confusing there.